tests/components/unifiprotect/test_init.py

```diff
@@ -186,15 +189,32 @@ async def test_setup_failed_error(hass: HomeAssistant, ufp: MockUFPFixture) -> N
 
 
 async def test_setup_failed_auth(hass: HomeAssistant, ufp: MockUFPFixture) -> None:
-    """Test setup of unifiprotect entry with unauthorized error."""
+    """Test setup of unifiprotect entry with unauthorized error after multiple retries."""
 
+    reauth_key = f"{ufp.entry.entry_id}_auth"
+    hass.data.setdefault(DOMAIN, {})[reauth_key] = AUTH_RETRIES
     ufp.api.get_nvr = AsyncMock(side_effect=NotAuthorized)
 
     await hass.config_entries.async_setup(ufp.entry.entry_id)
     assert ufp.entry.state == ConfigEntryState.SETUP_ERROR
     assert not ufp.api.update.called
 
 
+async def test_setup_failed_auth_retry(
+    hass: HomeAssistant, ufp: MockUFPFixture
+) -> None:
+    """Test setup of unifiprotect entry with unauthorized error."""
+
+    reauth_key = f"{ufp.entry.entry_id}_auth"
+    hass.data.setdefault(DOMAIN, {})[reauth_key] = 0
+    ufp.api.get_nvr = AsyncMock(side_effect=NotAuthorized)
+
+    await hass.config_entries.async_setup(ufp.entry.entry_id)
+    assert ufp.entry.state == ConfigEntryState.SETUP_RETRY
+    assert hass.data[DOMAIN][reauth_key] == 1
```

Comment:
HA integrations tests aren't allowed to access `hass.data`. I think we can avoid that by reloading the integration `AUTH_RETRIES` times and verifying the state moved from SETUP_RETRY to SETUP_ERROR